src/main/java/org/cyclops/integrateddynamics/core/evaluate/variable/ValueTypeListProxyMaterializedFactory.java

```diff
@@ -52,12 +57,20 @@ public String serialize(ValueTypeListProxyMaterialized<IValueType<IValue>, IValu
         if (valueType == null) {
             throw new IValueTypeListProxyFactoryTypeRegistry.SerializationException(String.format("Could not deserialize the serialized materialized list proxy value because the value type by name '%s' was not found.", valueTypeName));
         }
-        String[] values = new String[split.length - 1];
-        System.arraycopy(split, 1, values, 0, split.length - 1);
+        boolean heterogeneous = valueType.isCategory();
+        IValueType<IValue> elementValueType = valueType;
 
         ImmutableList.Builder<IValue> builder = ImmutableList.builder();
-        for (String serializedValue : values) {
-            IValue deserializedValue = valueType.deserialize(serializedValue.replaceAll(ELEMENT_DELIMITER_ESCAPED, ELEMENT_DELIMITER));
+        for (int i = 1; i < split.length; ++i) {
+            if (heterogeneous) {
```

Comment:
I think this will not be backwards-compatible with existing lists with category value types.

What we could do instead, is adding a flag to heterogeneous lists. If this flag is not available, deserialization code should be exactly the same as before (with a TODO that is should be removed in 1.13).

In 1.13, we can remove this flag and the backwards-compat code.

Reply:
How can any existing serialized lists have category value types? There's currently code in the serialize function that checks if a list has category type, and if so, pretends that it has the type of its first element. (And in the special case of an empty list, this isn't actually backwards incompatible.)

Reply:
Right, I forgot about that change (this wasn't always so).

I guess it's safe to assume that all lists were created after this change (it was quite a while ago), so no backwards-compat is needed here indeed.